### PR TITLE
Create markdown document for CVals

### PR DESCRIPTION
**Pull request by gbournou**
Create markdown document for CVals

b/280647950

### CODE REVIEW

**Review by kaidokert — APPROVED** [review]
This reads all great to me, i've taken a pass through and didn't spot anything off.

File: cobalt/doc/cvals.md [hunk 1]
```diff
@@ -0,0 +1,435 @@
+# Cobalt CVals
+
+## Overview
+
+CVals are globally accessible, thread-safe, key-value pairs within Cobalt, which
+are primarily used for monitoring state and tracking performance and memory.
+Each CVal is defined with a unique key and registered with the global
+CValManager. The CValManager can subsequently be queried for the current value
+of the key, which is returned as a string. CVals come in two varieties:
+PublicCVals, which are active in all builds; and DebugCVals, which are only
```
Comment (alexanderahn) [inline cobalt/doc/cvals.md:10]: Consider changing to list.

```suggestion
CVals come in two varieties:

- PublicCVals, which are active in all builds
- DebugCVals, which are only enabled in non-Gold builds
```

File: cobalt/doc/cvals.md [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+enabled in non-Gold builds.
+
+## Usage
+
+There are two ways to access CVals: through the Debug Console and via
```
Comment (alexanderahn) [inline cobalt/doc/cvals.md:15]: Same, consider changing to list.

File: cobalt/doc/cvals.md [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+
+### JavaScript
+
+CVals are exposed to JavaScript via the H5vccCVal object. This provides an
+interface for requesting all of the keys, which are returned as an array, and
+for retrieving a specific CVal value, which is returned as an optional string,
+via its CVal key. While the H5vccCVal object is usable in any build, only public
+CVals are queryable in Gold builds.
+
+Here are examples its usage:
```
Comment (alexanderahn) [inline cobalt/doc/cvals.md:44]:
```suggestion
Here are examples of its usage:
```

File: cobalt/doc/cvals.md [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+    existence globally. This includes ones that are pending garbage collection.
+*   **Count.DOM.Nodes** - The total number of Nodes in existence globally. This
+    includes ones that are pending garbage collection.
+*   **Count.MainWebModule.DOM.HtmlElement** - The total number of HtmlElements
+    in the MainWebModule. This includes elements that are not in the document
+    and ones that are pending garbage collection.
+*   **Count.MainWebModule.DOM.HtmlElement.Document** - The total number of
+    HtmlElements that are in the MainWebModule’s document.
+*   **Count.MainWebModule.DOM.HtmlScriptElement.Execute** - The total number of
+    HtmlScriptElement executions that have run since Cobalt started..
```
Comment (alexanderahn) [inline cobalt/doc/cvals.md:90]:
```suggestion
    HtmlScriptElement executions that have run since Cobalt started.
```

File: cobalt/doc/cvals.md [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+
+#### DebugCVals
+
+Media Pipeline exposes many State values, however we support multiple pipelines
+at once, so we have to query for the current pipeline number to access them.
+This can be done with `h5vcc.cVal.keys().filter(key =>
+key.startsWith("Media.Pipeline.") && key.endsWith("MaxVideoCapabilities") &&
+h5vcc.cVal.getValue(key).length === 0)` (If a Pipeline has MaxVideoCapabilities,
+it implies that the Pipeline is a 10x player, i.e. a secondary small player, so
+if MaxVideoCapabilities.length is 0, then it must be the primary pipeline. )
```
Comment (alexanderahn) [inline cobalt/doc/cvals.md:255]:
```suggestion
if MaxVideoCapabilities.length is 0, then it must be the primary pipeline.)
```

**Review by alexanderahn — APPROVED** [review]
Left a few minor suggestions and typo fixes.

**Pull request merged**